spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractMessageChannelBinder.java

```diff
@@ -224,18 +222,14 @@ protected void afterUnbindProducer(ProducerDestination destination, P producerPr
 			bridge.setBeanName("bridge." + name);
 			consumerEndpoint = createConsumerEndpoint(destination, group, properties);
 			consumerEndpoint.setOutputChannel(bridge);
-			if (consumerEndpoint instanceof InitializingBean) {
```

Comment:
A couple of reasons:

First problem arose when trying to get to split register and configure from the ErrorConfigurer. Because we were starting the components lifecycle before the configure had a chance to act. This is the whole reason, and it brings us closer to a model where we could have binding as a full fledge bean, I believe that once we have that, we can make it a lifecycle component and use proper start/stop hooks for the underlying components.